### ISSUE TITLE
CA1801 false positive

### ISSUE THREAD

**Issue by kentcb** [dotnet/roslyn-analyzers#1324]
#### Analyzer package

Microsoft.CodeAnalysis.FxCopAnalyzers

#### Analyzer

CA1801

#### Repro steps

```C#
public static readonly BindableProperty NameProperty =
    BindableProperty.Create(
        nameof(Name),
        typeof(string),
        typeof(Whatever),
        propertyChanged: OnNamePropertyChanged);

private static void OnNamePropertyChanged(BindableObject bindable, object oldValue, object newvalue)
{
}
```

#### Expected behavior

No violation because the method conforms to a delegate.

#### Actual behavior
```
warning CA1801: Parameter oldValue of method OnNamePropertyChanged is never used. Remove the parameter or use it in the method body.
```

**Comment by ivanbasov** [comment]
Thank you for reporting this!
The issue was fixed with [#1221](https://github.com/dotnet/roslyn-analyzers/pull/1221/)
It was not included in the currently shipped 2.3.0 beta. It is included in 2.6 which will be shipped after the IOperation update which is targeted by the end of 2017.

I have verified that the issue is gone with the analyzers code and the following sample:

```
using System;


namespace Repro
{
    public class Class1
    {
        public static readonly BindableProperty NameProperty =
            BindableProperty.Create(
                nameof(Name),
                typeof(string),
                typeof(Whatever),
                propertyChanged: OnNamePropertyChanged);

        private static void OnNamePropertyChanged(BindableObject bindable, object oldValue, object newvalue)
        {
        }
    }

    public class BindableObject
    {
    }

    public class BindableProperty
    {
        string _propertyName;
        Type _properyParameterType;
        Type _propertyWhateverType;
        Action<BindableObject, object, object> _propertyChanged

        public static BindableProperty Create(string propertyName, Type properyParameterType, Type propertyWhateverType, Action<BindableObject, object, object> propertyChanged)
        {
            var result = new BindableProperty();
            result._propertyName = propertyName;
            result._properyParameterType = properyParameterType;
            result.__propertyWhateverType = propertyWhateverType;
            result.__propertyChanged = propertyChanged;
            return result;
        }
    }

    public class Whatever { }

    public class Name { }
}
```
